step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java

```diff
@@ -68,6 +70,9 @@ public abstract class AutomationPackageManager {
     protected final ResourceManager resourceManager;
     protected final AutomationPackageHookRegistry automationPackageHookRegistry;
     protected boolean isIsolated = false;
+    protected final AutomationPackageLocks automationPackageLocks;
+
+    ExecutorService delayedUpdateExecutor = Executors.newFixedThreadPool(10);
```

Comment:
could be private and final.
not sure if shouldn't use a cached executor without limit

Reply:
did the change